### PR TITLE
fix(cloud manager): collect manager logs

### PR DESCRIPTION
**Pull request by juliayakovlev**
PR https://github.com/scylladb/siren-tests/pull/957 should be merged first

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/logcollector.py [hunk 1]
```diff
@@ -1222,6 +1222,35 @@ def get_gce_instances_by_testid(self):
...
+                LOGGER.error("Couldn't collect Siren manager logs, cluster_cloud module isn't installed")
+            else:
+                cloud_manager_id = self.params["cloud_manager_id"] if "cloud_manager_id" in self.params else None
+                LOGGER.info("GCE manager instance id: %s", cloud_manager_id)
+                if cloud_manager_id:
+                    cloud_cluster_id = self.params["cloud_cluster_id"]
+                    try:
+                        instance = get_gce_manager_instance_by_cluster_id(cluster_id=cloud_cluster_id)
+                        if not instance:
+                            raise ValueError("Cloud manager for the cluster %s not found", cloud_cluster_id)
```
Comment (enaydanov) [inline sdcm/logcollector.py:1239]: This is not a log message, please, use f-string here.

File: sdcm/logcollector.py [hunk 2]
```diff
@@ -1222,6 +1222,35 @@ def get_gce_instances_by_testid(self):
...
+                    except Exception as exc:
```
Comment (enaydanov) [inline sdcm/logcollector.py:1240]: Please, add pylint ignore here.

File: sdcm/logcollector.py [hunk 3]
```diff
@@ -1222,6 +1222,35 @@ def get_gce_instances_by_testid(self):
...
+                        LOGGER.error("Failed to get cloud manager instance. Error: %s", str(exc))
```
Comment (enaydanov) [inline sdcm/logcollector.py:1241]: No need to explicitly convert the exception to a string: `%s` in the format string do the same.

**Review by enaydanov — CHANGES_REQUESTED** [review]

File: sdcm/logcollector.py (hunk 3 above)
Comment (juliayakovlev) [reply sdcm/logcollector.py]: fixed

File: sdcm/logcollector.py (hunk 2 above)
Comment (juliayakovlev) [reply sdcm/logcollector.py]: fixed

File: sdcm/logcollector.py (hunk 1 above)
Comment (juliayakovlev) [reply sdcm/logcollector.py]: fixed

**Review by enaydanov — DISMISSED** [review]

File: sdcm/logcollector.py [hunk 4]
```diff
@@ -1222,6 +1222,35 @@ def get_gce_instances_by_testid(self):
...
                                                       global_ip=instance.public_ips[0],
                                                       tags={**self.tags, "NodeType": "loader", }))
+        if self.params["use_cloud_manager"]:
+            try:
+                from cluster_cloud import \
+                    get_gce_manager_instance_by_cluster_id  # pylint: disable=import-outside-toplevel
+            except ImportError:
+                LOGGER.error("Couldn't collect Siren manager logs, cluster_cloud module isn't installed")
+            else:
+                cloud_manager_id = self.params["cloud_manager_id"] if "cloud_manager_id" in self.params else None
```
Comment (fruch) [inline sdcm/logcollector.py:1232]: It easier just to use .get, self.params is a dictionary.
```suggestion
                cloud_manager_id = self.params.get("cloud_manager_id")
```

File: sdcm/logcollector.py [hunk 5]
```diff
@@ -1222,6 +1222,35 @@ def get_gce_instances_by_testid(self):
...
+                    LOGGER.info("GCE manager instance: %s", instance)
+                    ssh_login_info = {"hostname": instance["publicip"],
+                                      "user": "support",
+                                      "key_file": self.params["cloud_credentials_path"]}
+                    LOGGER.info("GCE manager instance ssh_login_info: %s", ssh_login_info)
+                    self.siren_manager_set.append(CollectingNode(name=instance["externalid"],
+                                                                 ssh_login_info=ssh_login_info,
+                                                                 instance=instance,
+                                                                 global_ip=instance["publicip"],
+                                                                 tags={"NodeType": "siren-manager", }))
```
Comment (fruch) [inline sdcm/logcollector.py:1253]: I'm not sure we should be tagging machines we are didn't create.

Do we have to pass down this parameter ?

File: sdcm/logcollector.py (hunk 4 above)
Comment (juliayakovlev) [reply sdcm/logcollector.py:1232]: "cloud_manager_id" is not defined config param. I don;t think we need to add it as parameter and set default value

File: sdcm/logcollector.py (hunk 5 above)
Comment (juliayakovlev) [reply sdcm/logcollector.py]: it's not mandatory. Removed

File: sdcm/logcollector.py (hunk 4 above)
Comment (fruch) [reply sdcm/logcollector.py:1232]: as in any dict, we don't need it defined. `dict.get` would return None if the key doesn't exist.

Comment (juliayakovlev) [reply sdcm/logcollector.py]: Changed. I didn't run the test

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**